### PR TITLE
e2e smoke test

### PR DESCRIPTION
**Pull request by olga-mir**
### Which issue this PR addresses:

First part of https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/7361152/

### What this PR does / why we need it:

Migrating basic e2e smoke tests from v3. https://github.com/openshift/openshift-azure/blob/master/test/e2e/standard/cluster.go

This PR includes 2 checks from v3 repo:
Persistent Volumes: https://github.com/openshift/openshift-azure/blob/4205d734414be919df805c5599743a17715cfc65/test/e2e/standard/cluster.go#L365-L453
Load Balancer services: https://github.com/openshift/openshift-azure/blob/4205d734414be919df805c5599743a17715cfc65/test/e2e/standard/cluster.go#L242-L272

There are more high level tests: https://github.com/openshift/openshift-azure/blob/master/test/e2e/standard/testapps.go
These are usng templates, but I want to re-work them to use OLM instead, the prep work and code for this is mostly done, but I need a good candidate for the operator to test on: https://github.com/Azure/ARO-RP/compare/master...olga-mir:smoke-test?expand=1

### Test plan for issue:

`make e2e`

### Is there any documentation that needs to be updated for this PR?

no

### CODE REVIEW

**Comment by ehashman** [comment]
@olga-mir to make this easier to review, can you compare what's been ported here to what's in the v3 repo? I'm not sure what the scope of this PR should be.

**Comment by olga-mir** [comment]
@ehashman this PR includes 2 checks from v3 repo:
Persistent Volumes: https://github.com/openshift/openshift-azure/blob/4205d734414be919df805c5599743a17715cfc65/test/e2e/standard/cluster.go#L365-L453
Load Balancer services: https://github.com/openshift/openshift-azure/blob/4205d734414be919df805c5599743a17715cfc65/test/e2e/standard/cluster.go#L242-L272

There are more high level tests: https://github.com/openshift/openshift-azure/blob/master/test/e2e/standard/testapps.go
These are usng templates, but I want to re-work them to use OLM instead, the prep work and code for this is mostly done, but I need a good candidate for the operator to test on: https://github.com/Azure/ARO-RP/compare/master...olga-mir:smoke-test?expand=1

**Comment by ehashman** [comment]
Thanks for the additional context -- I updated the PR description accordingly :)

File: test/e2e/cluster.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+		}).Should(BeNil())
+	})
+
+	var _ = AfterEach(func() {
+		err := project.Delete()
+		Expect(err).NotTo(HaveOccurred(), "Failed to delete test namespace")
+
+		Eventually(func() error {
+			return project.VerifyProjectIsDeleted()
+		}, 5*time.Minute, 1*time.Second).Should(BeNil())
```
Comment (asalkeld) [inline test/e2e/cluster.go:48]:
```suggestion
		}, 5*time.Minute, 10*time.Second).Should(BeNil())
```

File: test/e2e/cluster.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+					},
+				},
+			},
+			RestartPolicy: corev1.RestartPolicyNever,
+		},
+	})
+	return err
+}
+
+func verifyPodSucceeded(cli kubernetes.Interface) error {
```
Comment (asalkeld) [inline test/e2e/cluster.go:154]: https://github.com/Azure/ARO-RP/blob/master/test/util/ready/ready.go#L23 ?

File: test/e2e/cluster.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+				},
+			},
+			Type: corev1.ServiceTypeLoadBalancer,
+		},
+	}
+	_, err := cli.CoreV1().Services(testNamespace).Create(svc)
+	return err
+}
+
+func verifyServiceIsReady(ctx context.Context, cli kubernetes.Interface, svcName string) error {
```
Comment (asalkeld) [inline test/e2e/cluster.go:186]: maybe move to the test/util/ready module?

File: test/e2e/setup.go [hunk 4]
```diff
@@ -85,11 +82,16 @@ func newClientSet(subscriptionID string) (*clientSet, error) {
...
 		MachineAPI: machineapicli,
+		Project:    projectv1client.NewForConfigOrDie(restConfig),
 	}, nil
 }
 
 var _ = BeforeSuite(func() {
 	log.Info("BeforeSuite")
+
+	SetDefaultEventuallyTimeout(2 * time.Minute)
+	SetDefaultEventuallyPollingInterval(1 * time.Second)
```
Comment (asalkeld) [inline test/e2e/setup.go:93]:
```suggestion
	SetDefaultEventuallyPollingInterval(10 * time.Second)
```

File: test/e2e/cluster.go (hunk 2 above)
Comment (olga-mir) [reply test/e2e/cluster.go:154]: @asalkeld this pod finishes quickly so we risk to miss "ready" state. That's why it is checking for Succeeded instead

**Pull request merged**